### PR TITLE
feat(ui): Native stack frame renderer enhancements

### PR DESCRIPTION
**Pull request by matejminar**
Updates to the native stack frame renderer:
- convert between relative/absolute addresses
- underline address if found by scanning
- package path tooltip instead of slower title
- clicking on package scrolls down to the filtered images
- relative addresses by default
- images search box refactored to reusable component with icon and reset option
- status icons
- inlined frame indicator

Screenshot is filled with dummy data, so it might not make sense - it's just for demonstration purposes only:
Screenshot 2019-12-03 at 12 46 14

### CODE REVIEW

**Review by ckj — CHANGES_REQUESTED** [review]
Overall, this looks great @matejminar. I would like to request an alignment/padding fix regarding filter box here:

images-loaded-alignment

Can you make the filter box size and alignment match the one from the breadcrumbs section? 

breadcrumbs-alignment

**Comment by matejminar** [comment]
Thank you @ckj , I've fixed it.

**Comment by matejminar** [comment]
This is the latest screenshot:
Screenshot 2019-12-05 at 16 13 20
@markstory can you please review the code and @ckj would you please take a look if everything looks fine?

**Review by ckj — APPROVED** [review]
Design-wise, I think we're good to go 💥

File: src/sentry/static/sentry/app/components/events/interfaces/togglableAddress.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  startingAddress: string | null;
+  isAbsolute: boolean;
+  onToggle?: () => void;
+  isFoundByStackScanning: boolean;
+  isInlineFrame: boolean;
+  maxLengthOfRelativeAddress: number;
+};
+
+class TogglableAddress extends React.Component<Props> {
+  convertAbsoluteAddressToRelative = () => {
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/togglableAddress.tsx]: Does this need to be an arrow function? You aren't using it as an event handler.

File: src/sentry/static/sentry/app/components/events/interfaces/togglableAddress.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    const relativeAddress = formatAddress(
+      parseAddress(address) - parseAddress(startingAddress),
+      maxLengthOfRelativeAddress
+    );
+
+    return `+${relativeAddress}`;
+  };
+
+  getAddressTooltip = () => {
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/togglableAddress.tsx]: Does this need to be an arrow function? You aren't using it as an event handler.

**Review by markstory — COMMENTED** [review]
Is the UI settled down enough that you could add some jest tests to ensure filtering and other operations don't regress?

File: src/sentry/static/sentry/app/components/events/interfaces/debugmeta.jsx [hunk 3]
```diff
@@ -10,33 +10,23 @@ import Checkbox from 'app/components/checkbox';
...
-    return 0;
-  }
-}
-
 function getImageRange(image) {
   // The start address is normalized to a `0x` prefixed hex string. The event
   // schema also allows ingesting plain numbers, but this is converted during
   // ingestion.
-  const startAddress = parseAddr(image.image_addr);
+  const startAddress = parseAddress(image.image_addr, IMAGE_ADDR_LEN);
```
[the remark this answers is not in the document]
Comment (matejminar) [reply src/sentry/static/sentry/app/components/events/interfaces/debugmeta.jsx]: You are right,  I've mistaken it for formatAddress which takes two. Need that TS coverage in my life.

File: src/sentry/static/sentry/app/components/events/interfaces/imageForBar.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import space from 'app/styles/space';
+import {t} from 'app/locale';
+
+type Props = {
+  frame: object;
+  onShowAllImages: () => void;
+};
+
+class ImageForBar extends React.Component<Props> {
+  render() {
```
[the remark this answers is not in the document]
Comment (matejminar) [reply src/sentry/static/sentry/app/components/events/interfaces/imageForBar.tsx]: 👍 changed, thanks - I wonder whether we could disable react/prop-types eslint rule in .tsx files, so we don't have to specify typescript types AND proptypes... I think it works fine with typescript class components, but in functional components, compiler complained

File: src/sentry/static/sentry/app/components/events/interfaces/togglableAddress.tsx (hunk 1 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/components/events/interfaces/togglableAddress.tsx]: No, thank you, I've changed it. Just out of curiosity, is there some downside with writing all methods as arrow functions? Or is it less expressive?

File: src/sentry/static/sentry/app/components/events/interfaces/frame.jsx [hunk 5]
```diff
@@ -551,6 +631,16 @@ const NativeLineContent = styled(RepeatedContent)`
 
 const DefaultLine = styled(VertCenterWrapper)`
   justify-content: space-between;
+
+  &:hover ${PackageStatusIcon} {
+    opacity: 1;
+  }
+`;
+
+const HintStatus = styled(InlineSvg)`
```
[the remark this answers is not in the document]
Comment (matejminar) [reply src/sentry/static/sentry/app/components/events/interfaces/frame.jsx]: I wish, but this is still a .jsx file. Will have to refactor to .tsx sometimes in the future.

**Review by markstory — APPROVED** [review]

**Review by jan-auer — APPROVED** [review]
🎉 Thanks, this is going to be a great improvement over what we have now!

**Pull request merged**